src/lib/Form/Type/Content/BaseContentType.php

```diff
@@ -106,6 +101,20 @@ public function configureOptions(OptionsResolver $resolver)
                 'ibexa/content-forms',
                 'v4.6.4',
                 'The option "%name%" is deprecated, use "struct" instead.'
+            )
+            ->setNormalizer(
+                'struct',
+                static function (Options $options, ?ContentStruct $value): ?ContentStruct {
+                    if ($value === null) {
+                        trigger_deprecation(
+                            'ibexa/content-forms',
+                            'v4.6',
+                            'The option "struct" with null value is deprecated and will be required in v5.0.'
+                        );
+                    }
+
+                    return $value;
+                }
```

Comment:
Are we sure normalizer runs before `default` is resolved? I'm pretty sure we specifically did not use normalizer for options, because this happens after defaults callback is called.

It could be put in the defaults callback instead.

Reply:
It worked properly when I tested it, but maybe you are addressing some other case here, if you could be more specific about when this could fail I can test it again.

Reply:
If it works, then no issue from my end. I remember I talked with the author who insisted that this normalizer didn't work when `struct` option was left untouched.

Apparently, Symfony Form complained that `struct` option is required, but is not set at all. Maybe it was just the order of operation that impacted it, i.e.:
 * Field is set as required
 * There is no default (value nor callback)
 * We never reach normalizer, because there is no default and option is required

I would say that maybe option defaults is a better place for this code, as this "happens" then `struct` is `null` and we are using one of the old options. 

If `struct` is simply `null` and there are not other options, Symfony Form will complain that it's missing anyway.